pkg/istio-agent/xds_proxy.go

```diff
@@ -728,19 +728,11 @@ func (p *XdsProxy) getTLSDialOption(agent *Agent) (grpc.DialOption, error) {
 	// strip the port from the address
 	parts := strings.Split(agent.proxyConfig.DiscoveryAddress, ":")
 	config.ServerName = parts[0]
-
-	// For debugging on localhost (with port forward)
-	// This matches the logic for the CA; this code should eventually be shared
-	if strings.Contains(config.ServerName, "localhost") {
-		config.ServerName = "istiod.istio-system.svc"
-	}
-
 	if p.istiodSAN != "" {
 		config.ServerName = p.istiodSAN
 	}
 	// TODO: if istiodSAN starts with spiffe://, use custom validation.
 
-	config.MinVersion = tls.VersionTLS12
```

Comment:
why remove this? because it is default?

Reply:
I second that this line should not be removed. It is more straightforward and less error-prone to explicitly configure the min version. Thus, in case the default in Envoy change, the Istio behavior remains stable and unchanged.

Reply:
Yes